app/src/main/java/org/mozilla/fenix/perf/Performance.kt

```diff
@@ -37,4 +45,59 @@ object Performance {
             activity.reportFullyDrawn()
         }
     }
+
+    /**
+     * Processes intent for Performance testing to remove protection pop up ( but keeps the TP
+     * on) and removes the onboarding screen.
+     */
+    fun processIntentIfPerformanceTest(intent: Intent, context: Context) {
+        if (isPerformanceTest(intent)) {
+            return
+        }
+
+        if (evaluatePhoneStatus(context)) {
+            disableOnboarding(context)
+            disableTrackingProtectionPopups(context)
+        }
+    }
+
+    /**
+     * Checks if the performance Intent was received for performance testing.
+     */
+    fun isPerformanceTest(intent: Intent): Boolean {
+        return !intent.getBooleanExtra(EXTRA_IS_PERFORMANCE_TEST, false)
+    }
+
+    /**
+     * The test will be allowed to run if the phone is connected through usb and if it has
```

Comment:
This doesn't really describe what this method does – it describes how it's being used, which you should generally try to avoid because it's not colocated with the code that calls it and is very likely to get outdated. In general, I'm not sure this comment is needed.